packages/lesswrong/server/callbacks/userCallbackFunctions.tsx

```diff
@@ -352,59 +357,73 @@ export async function updateDigestSubscription(data: Partial<DbUser>, {oldDocume
       throw err
     }
     data.subscribedToDigest = false
+    data.subscribedToNewsletter = false
     return data;
   }
 
-  if (
-    isAnyTest ||
-    !hasDigests ||
-    data.subscribedToDigest === undefined || // When a mutation doesn't reference subscribedToDigest
+  const noDigestUpdate = !hasDigests ||
+    data.subscribedToDigest === undefined ||
     data.subscribedToDigest === oldDocument.subscribedToDigest
-  ) {
+  const noNewsletterUpdate = !hasNewsletter ||
+    data.subscribedToNewsletter === undefined ||
+    data.subscribedToNewsletter === oldDocument.subscribedToNewsletter
+  if (isAnyTest || (noDigestUpdate && noNewsletterUpdate)) {
     return data;
   }
 
   const mailchimpAPIKey = mailchimpAPIKeySetting.get();
   const mailchimpForumDigestListId = mailchimpForumDigestListIdSetting.get();
-  if (!mailchimpAPIKey || !mailchimpForumDigestListId) {
-    return handleErrorCase("Error updating digest subscription: Mailchimp not configured")
+  const mailchimpEANewsletterListId = mailchimpEAForumNewsletterListIdSetting.get();
+  const digestUpdateButNoListId = hasDigests && !mailchimpForumDigestListId
+  const newsletterUpdateButNoListId = hasNewsletter && !mailchimpEANewsletterListId
+  if (!mailchimpAPIKey || digestUpdateButNoListId || newsletterUpdateButNoListId) {
```

Comment:
```suggestion
  if (!mailchimpAPIKey || (digestUpdateButNoListId && newsletterUpdateButNoListId)) {
```
It should just skip if one is missing

Reply:
Kinda. It should still record an error.